klangbecken.liq

```diff
@@ -56,23 +56,80 @@ ALSA_DEVICE = getenv_fallback("KLANGBECKEN_ALSA_DEVICE", "default")
 # PLAYLISTS                                         #
 # ================================================= #
 
+# parse index.json and keep it up to date
+index = ref []
+index_filename = path.concat(DATA_DIR, "index.json")
+def read_index() =
+  log("updating index", label="update_index")
+  index := of_json(default=[("list", [("key", "value")])], file.contents(index_filename))
+end
+if file.exists(index_filename) then read_index() end
+file.watch(index_filename, read_index)
+
+# calculate waiting time for repeating a track depending on its playlist
+def calc_wait(playlist) =
+  if playlist == "music" then 259200 # 3 days
+  elsif playlist == "classics" then 86400 # 1 day
+  elsif playlist == "jingles" then 3600 # 1 hour
```

Comment:
from what i understood there are different frequencies for jingles. do we handle this somewhere else and this is just an additional check?

Reply:
This is done further down in the [script](https://github.com/radiorabe/klangbecken/blob/496c20f80f781efdbc3bc99630b8a83d781d5080/klangbecken.liq#L153-L182), where the playlists are put together. The function here is for not repeating tracks in their own playlist (music, classics and jingles).